src/main/java/com/cedarsoftware/util/reflect/ReflectionUtils.java

```diff
@@ -81,18 +63,29 @@ public static Map<String, Method> buildInjectorMap(Class<?> c) {
      * @param method2 - 2nd method to compare
      * @return in the case over overloads choose the method with 0 parameters.
      */
-    public static Method zeroParameterMethodPreference(Method method1, Method method2) {
-        return method1.getParameterCount() == 0 ? method1 : method2;
+    public static Method oneParameterMethodPreference(Method method1, Method method2) {
+        return method1.getParameterCount() == 1 ? method1 : method2;
+    }
+
```

Comment:
Would it be beneficial to have access to all methods, and having non-collided with a key of method name + (param type names)... so that all methods are considered?